File: tests/rust/tests/signal_filtering.rs
use brainflow::board_shim;
use brainflow::data_filter;
use brainflow::BoardIds;
use brainflow::FilterTypes;

mod common;

#[test]
fn signal_filtering() {
    let board_id = BoardIds::SyntheticBoard as i32;
    let sampling_rate = board_shim::get_sampling_rate(board_id).unwrap();
    let eeg_channels = board_shim::get_eeg_channels(board_id).unwrap();

    let mut data = common::board_data(Some(10));

    println!("{:?}", data[eeg_channels[0]]);

    data_filter::perform_lowpass(
        &mut data[eeg_channels[0]],
        sampling_rate,
        13.0,
        3,
        FilterTypes::Bessel as i32,
        0.0,
    )
    .unwrap();

    println!("{:?}", data[eeg_channels[0]]);
}

File: matlab_package/brainflow/examples/SignalFiltering.m
BoardShim.set_log_file('brainflow.log');
BoardShim.enable_dev_board_logger();

params = BrainFlowInputParams();
board_shim = BoardShim(int32(BoardIds.SYNTHETIC_BOARD), params);
preset = int32(BrainFlowPresets.DEFAULT_PRESET);
board_shim.prepare_session();
board_shim.start_stream(45000, '');
pause(5);
board_shim.stop_stream();
data = board_shim.get_current_board_data(64, preset);
board_shim.release_session();

eeg_channels = BoardShim.get_eeg_channels(int32(BoardIds.SYNTHETIC_BOARD), preset);
% apply iir filter to the first eeg channel %
first_eeg_channel = eeg_channels(1);
original_data = data(first_eeg_channel, :);
sampling_rate = BoardShim.get_sampling_rate(int32(BoardIds.SYNTHETIC_BOARD), preset);
filtered_data = DataFilter.perform_lowpass(original_data, sampling_rate, 50.0, 3, int32(FilterTypes.BUTTERWORTH), 0.0);